ui/libs/layout/src/lib/components/main-header/utils.ts

```diff
@@ -1,21 +1,17 @@
-import { catchError, Observable, of, tap } from 'rxjs';
-import {EoscCommonWindow} from "@eosc-search-service/common";
+import { delay, Observable, tap } from 'rxjs';
+import { EoscCommonWindow, UserProfile } from '@eosc-search-service/common';
 
 declare let window: EoscCommonWindow;
 
-export const rerenderComponent = (
+export function rerenderComponent$(
   id: string,
-  userInfo$: Observable<{ username: string }>
-) => {
+  userInfo$: Observable<UserProfile | null>
+) {
   return userInfo$.pipe(
-    catchError(() => {
-      window.eosccommon.renderMainHeader(`#${id}`);
-      return of();
-    }),
-    tap(({ username }) =>
-      window.eosccommon.renderMainHeader(`#${id}`, {
-        username,
-      })
+    // required for rerendering to work
+    delay(0),
```

Comment:
why are you using the `delay`. I believe it's some kind of async hack to put user update to the end of the stack, but I may be wrong.

Reply:
Exactly - this is equivalent to having instructions wrapped in `setTimeout` they will be executed after current instruction pass. Here it is required because rerendering basically uses another framework so we make sure that everything is ready and only then force the rerender.